nondex-instrumentation/src/main/java/edu/illinois/nondex/instr/HashIteratorShufflerASMDump.java

```diff
@@ -70,6 +70,14 @@
             mv.visitMethodInsn(INVOKESTATIC, "edu/illinois/nondex/shuffling/ControlNondeterminism", "shuffle", "(Ljava/util/List;)Ljava/util/List;", false);
             mv.visitVarInsn(ASTORE, 3);
             mv.visitVarInsn(ALOAD, 0);
+            mv.visitFieldInsn(GETFIELD, baseName + "$HashIterator$HashIteratorShuffler", "hashIter", "L" + baseName + "$HashIterator;");
+            mv.visitInsn(ACONST_NULL);
+            if("java/util/WeakHashMap".equals(baseName)){
+                mv.visitFieldInsn(PUTFIELD, baseName + "$HashIterator", "lastReturned", "L" + baseName + "$" + type + ";");
```

Comment:
You do not need this if. Look in Instrumenter.java on line 140 it invokes the dump(...) method with certain arguments. There, the variable name is passed in, i.e., current or entry. I believe you need to change that call, in the branch corresponding to WeakHashMap to call with "lastReturned". That way you'll not have this if here but rather just pass from outside the right argument.